src/types/messages.d.ts

```diff
@@ -5,10 +5,17 @@ interface IMessageJSONBase {
   sender: IPublicAccount;
   recipient: string;
   timestamp: Date | string;
+  messageInfo?: MessageInfo;
   signature?: string;
   hash?: string;
 }
 
+interface MessageInfo {
```

Comment:
Interfaces must start with `I`, so `IMessageInfo`.
Do we really need an interface, instead of just setting the properties of `IMessage`?

How are you differentiating between an empty string and `undefined` when from/to binary? Better make both title and descripton required.
An empty thumbnail should always be interpreted as no thumbnail.